litrpc/lit-accounts.proto

```diff
@@ -0,0 +1,142 @@
+syntax = "proto3";
+
+package litrpc;
+
+option go_package = "github.com/lightninglabs/lightning-terminal/litrpc";
+
+service Accounts {
+    /*
+    CreateAccount adds an entry to the account database. This entry represents
+    an amount of satoshis (account balance) that can be spent using off-chain
+    transactions (e.g. paying invoices).
+
+    Macaroons can be created to be locked to an account. This makes sure that
+    the bearer of the macaroon can only spend at most that amount of satoshis
+    through the daemon that has issued the macaroon.
+
+    Accounts only assert a maximum amount spendable. Having a certain account
+    balance does not guarantee that the node has the channel liquidity to
+    actually spend that amount.
+    */
+    rpc CreateAccount (CreateAccountRequest) returns (CreateAccountResponse);
+
+    /*
+    UpdateAccount updates an existing account in the account database.
+    */
+    rpc UpdateAccount (UpdateAccountRequest) returns (Account);
+
+    /*
+    ListAccounts returns all accounts that are currently stored in the account
+    database.
+    */
+    rpc ListAccounts (ListAccountsRequest) returns (ListAccountsResponse);
+
+    /*
+    RemoveAccount removes the given account from the account database.
+    */
+    rpc RemoveAccount (RemoveAccountRequest) returns (RemoveAccountResponse);
+}
+
+message CreateAccountRequest {
+    /*
+    The initial account balance in satoshis representing the maximum amount that
+    can be spent.
+    */
+    uint64 account_balance = 1;
+
+    /*
+    The expiration date of the account as a timestamp. Set to 0 to never expire.
+    */
+    int64 expiration_date = 2;
+}
+
+message CreateAccountResponse {
+    // The new account that was created.
+    Account account = 1;
+
+    // The macaroon with all permissions required to access the account.
+    bytes macaroon = 2;
+}
+
+message Account {
+    // The ID of the account.
+    string id = 1;
+
+    /*
+    The initial balance in satoshis that was set when the account was created.
+    */
+    uint64 initial_balance = 2;
+
+    // The current balance in satoshis.
+    int64 current_balance = 3;
+
+    // Timestamp of the last time the account was updated.
+    int64 last_update = 4;
+
+    /*
+    Timestamp of the account's expiration date. Zero means it does not expire.
+    */
+    int64 expiration_date = 5;
+
+    /*
+    The list of invoices created by the account. An invoice created by an
+    account will credit the account balance if it is settled.
+    */
+    repeated AccountInvoice invoices = 6;
+
+    /*
+    The list of payments made by the account. A payment made by an account will
+    debit the account balance if it is settled.
+    */
+    repeated AccountPayment payments = 7;
```

Comment:
One last thought when I was looking through this PR, and also probably something that could be addressed in a follow-up PR, but it looks like these include all payments and invoices ever made by the accounts. Therefore when calling `ListAccounts`, you're fetching all payment and invoice history for all accounts across all time. That seems like it could potentially be a very heavy query.

It might make sense to have a `ListAccounts` call that just gives you the accounts and balances that exist, and then separate `ListInvoices` and `ListPayments` calls that work for a single account specified in the request? Something like that, just so that there's a way to list the accounts without needing to fetch all invoice & payment history.

I can put this in an issue as well, just wanted to raise it here first.